mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergRecordWriter.java

```diff
@@ -0,0 +1,106 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.hadoop.hive.ql.exec.FileSinkOperator;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.Reporter;
+import org.apache.hadoop.mapred.TaskAttemptID;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionKey;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.io.FileAppenderFactory;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.OutputFileFactory;
+import org.apache.iceberg.io.PartitionedFanoutWriter;
+import org.apache.iceberg.mr.mapred.Container;
+import org.apache.iceberg.util.Tasks;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class HiveIcebergRecordWriter extends PartitionedFanoutWriter<Record>
+    implements FileSinkOperator.RecordWriter, org.apache.hadoop.mapred.RecordWriter<NullWritable, Container<Record>> {
+  private static final Logger LOG = LoggerFactory.getLogger(HiveIcebergRecordWriter.class);
+
+  // The current key is reused at every write to avoid unnecessary object creation
+  private final PartitionKey currentKey;
+  private final FileIO io;
+
+  // <TaskAttemptId, HiveIcebergRecordWriter> map to store the active writers
+  // Stored in concurrent map, since some executor engines can share containers
+  private static final Map<TaskAttemptID, HiveIcebergRecordWriter> writers = new ConcurrentHashMap<>();
+
+  static HiveIcebergRecordWriter removeWriter(TaskAttemptID taskAttemptID) {
+    return writers.remove(taskAttemptID);
+  }
+
+  HiveIcebergRecordWriter(Schema schema, PartitionSpec spec, FileFormat format,
+      FileAppenderFactory<Record> appenderFactory, OutputFileFactory fileFactory, FileIO io, long targetFileSize,
+      TaskAttemptID taskAttemptID) {
+    super(spec, format, appenderFactory, fileFactory, io, targetFileSize);
+    this.io = io;
+    this.currentKey = new PartitionKey(spec, schema);
+    writers.put(taskAttemptID, this);
+  }
+
+  @Override
+  protected PartitionKey partition(Record row) {
+    currentKey.partition(row);
+    return currentKey;
+  }
+
+  @Override
+  public void write(Writable row) throws IOException {
+    super.write(((Container<Record>) row).get());
+  }
+
+  @Override
+  public void write(NullWritable key, Container value) throws IOException {
+    write(value);
+  }
+
+  @Override
+  public void close(boolean abort) throws IOException {
+    DataFile[] dataFiles = super.complete();
```

Comment:
This was just changed in #1867. I think you just need to update it to `super.dataFiles()`.

Reply:
Fixed
Thanks for the pointer!